internals/secrethub/org_rm.go

```diff
@@ -71,3 +74,12 @@ func (cmd *OrgRmCommand) Run() error {
 
 	return nil
 }
+
+func (cmd *OrgRmCommand) argumentRegister(c *cobra.Command, args []string) error {
```

Comment:
Ideally we would have a method on the base command type (`cli.CommandClause`) for registering arguments, just like we the methods for flags. It could take a `Value` type as a parameter, or just a pointer to a string as currently all args are handled as strings, if I'm not mistaken.

Reply:
I believe the way that it is done now has been our collective design decision. If you remember, I, you and @edif2008 approached together the problem of 'transferring' the arguments from the `args` array to the variables that we use and decided that this should be done in the PreRun routine of the command. Therefore, the function should abide by this function signature. Moreover, this function, at the moment, transfers more than one argument from the array. Providing pointers to strings as a parameter (in an array, for example), would introduce, from my point of view, unnecessary complexity (adding the pointers to an array, checks for sizes, needing to ensure a specific order for the pointers in the array etc.) that we circumvent, by doing it like this. Please, let me know if I'm missing something.

Reply:
I should have provided a more detailed explanation here :sweat_smile: 

The collective decision we took was to handle parsing the arguments in the `PreRun` method of the command and I still stand by that decision.

My remark is regarding the way this was implemented. Having a function per command for parsing arguments is a good first step in this transition, but I think the amount of code required can be significantly reduced.

My current proposal is to have a method on `cli.CommandClause` that takes a list of string pointers and modifies the `PreRun` of the given command to transfer the argument values to these pointers. This would require one line of code for each command that uses arguments.

I also agree that using a slice of pointers in this case could be quite cumbersome thanks for pointing it out :+1:  In this case we could use [variadic functions](https://gobyexample.com/variadic-functions). Do you think these would solve most of the complexity you mentioned?

Also message me if you have more questions or ideas about this topic specifically, having a call would be the quickest way of discussing them.

Reply:
We discussed this and agreed to attempt to implement it.

Reply:
This has been done. Let us know if you have any more suggestions related to this, otherwise feel free to close the thread. :D

Reply:
Can this conversation be resolved?